connectivity/check/deployment.go

```diff
@@ -621,13 +632,25 @@ func (ct *ConnectivityTest) waitForService(ctx context.Context, service Service)
 
 		// Warning: ExecInPodWithStderr ignores ctx. Don't pass it here so we don't
 		// falsely expect the function to be able to be cancelled.
-		_, e, err := ct.client.ExecInPodWithStderr(context.TODO(),
+		stdout, e, err := ct.client.ExecInPodWithStderr(context.TODO(),
 			pod.Pod.Namespace, pod.Pod.Name, pod.Pod.Labels["name"],
 			[]string{"nslookup", service.Service.Name}) // BusyBox nslookup doesn't support any arguments.
 
 		// Lookup successful.
 		if err == nil {
-			return nil
+			svcIP := ""
+			switch service.Service.Spec.Type {
+			case corev1.ServiceTypeClusterIP, corev1.ServiceTypeNodePort:
+				svcIP = service.Service.Spec.ClusterIP
+			case corev1.ServiceTypeLoadBalancer:
+				if len(service.Service.Status.LoadBalancer.Ingress) > 0 {
+					svcIP = service.Service.Status.LoadBalancer.Ingress[0].IP
+				}
+			}
+			if strings.Contains(stdout.String(), svcIP) {
```

Comment:
Could we maybe check for the full `Address: <IP>` line here? In its current state the check would mistakenly succeed e.g. for IP `192.168.1.1` if the address in `nslookup` output was in fact `192.168.1.10`.

(I think @pchaigno recently fixed a test flake in `cilium/cilium` caused by a similar piece of code)

Reply:
Good point, done :-)